src/python_testing/TC_SC_3_6.py

```diff
@@ -0,0 +1,174 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from unicodedata import name
+from matter_testing_support import MatterBaseTest, default_matter_test_main, async_test_body
+from chip.interaction_model import Status
+import chip.clusters as Clusters
+import chip.FabricAdmin
+import logging
+from mobly import asserts
+from chip.utils import CommissioningBuildingBlocks
+from chip.clusters.Attribute import TypedAttributePath, SubscriptionTransaction
+import asyncio
+from threading import Event
+import time
+
+
+class ResubscriptionCatcher:
+    def __init__(self, name):
+        self._name = name
+        self._got_resubscription_event = Event()
+
+    def __call__(self, transaction: SubscriptionTransaction, terminationError, nextResubscribeIntervalMsec):
+        self._got_resubscription_event.set()
+        logging.info("Got resubscription on client %s" % self.name)
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def caught_resubscription(self) -> bool:
+        return self._got_resubscription_event.is_set()
+
+
+class TC_SC_3_6(MatterBaseTest):
+    @async_test_body
+    async def test_TC_SC_3_6(self):
+        dev_ctrl = self.default_controller
+
+        # Get overrides for debugging the test
+        num_fabrics_to_commission = self.user_params.get("num_fabrics_to_commission", 5)
+        num_controllers_per_fabric = self.user_params.get("num_controllers_per_fabric", 3)
+        min_report_interval_sec = self.user_params.get("min_report_interval_sec", 1)
+        max_report_interval_sec = self.user_params.get("max_report_interval_sec", 30)
```

Comment:
With such a large max interval, the re-sub logic is not actually going to fire, meaning your logic below that checks for a re-sub event is not actually going to fire, _even if_ the subscription actually dies on the server.

I'm not sure how exactly we should deal with this without prolonging the test a whole lot. Need to think...